### ISSUE TITLE
error: call to function 'operator<<' that is neither visible in the template definition nor found by argument-dependent lookup

### ISSUE THREAD

**Issue by seanfarley** [djcb/mu#1550]
## Expected or desired behavior

make is not working

## Actual behavior

```
$ make
/Applications/Xcode.app/Contents/Developer/usr/bin/make  all-recursive
Making all in m4
make[2]: Nothing to be done for `all'.
Making all in man
/Applications/Xcode.app/Contents/Developer/usr/bin/make  all-am
make[3]: Nothing to be done for `all-am'.
Making all in lib
/Applications/Xcode.app/Contents/Developer/usr/bin/make  all-recursive
Making all in utils
  CXX      mu-command-parser.lo
In file included from mu-command-parser.cc:20:
In file included from ./mu-command-parser.hh:29:
In file included from ../../lib/utils/mu-error.hh:25:
./mu-utils.hh:151:14: error: call to function 'operator<<' that is neither visible in the template definition nor found by argument-dependent lookup
        sstr << val;
             ^
mu-command-parser.cc:81:45: note: in instantiation of function template specialization 'Mu::to_string<Mu::Sexp::Type>' requested here
                                            to_string(arginfo.type) +
                                            ^
../../lib/utils/mu-sexp-parser.hh:83:1: note: 'operator<<' should be declared prior to the call site or in namespace 'Mu::Sexp'
operator<<(std::ostream& os, Sexp::Type id)
^
1 error generated.
make[4]: *** [mu-command-parser.lo] Error 1
make[3]: *** [all-recursive] Error 1
make[2]: *** [all] Error 2
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
```

## Steps to reproduce

`./autogen.sh && ./configure && make`

## Versions of mu, mu4e/emacs, operating system etc.

mu: 5a5cb014b648ec513d6cda4bd4d882a07f2b4615
OS: MacOS Mojave

## Any other detail

I bisected with git and the breaking commit seems to be c7680b78b930462194c8d30bad987e9e30ea9d8b

**Comment by seanfarley** [comment]
Still happening after 72e67ca389a62b5538fd8a373badcc3b001b9ef7 :-(

```
make                                                                                                                                                                           
/Applications/Xcode.app/Contents/Developer/usr/bin/make  all-recursive
Making all in m4
make[2]: Nothing to be done for `all'.
Making all in man
cp -f mu-find.1 mu-mfind.1
/Applications/Xcode.app/Contents/Developer/usr/bin/make  all-am
make[3]: Nothing to be done for `all-am'.
Making all in lib
touch dummy.cc
/Applications/Xcode.app/Contents/Developer/usr/bin/make  all-recursive
Making all in utils
  CC       test-mu-util.o
  CC       mu-date.lo
  CC       mu-log.lo
  CXX      mu-command-parser.lo
In file included from mu-command-parser.cc:20:
In file included from ./mu-command-parser.hh:29:
In file included from ../../lib/utils/mu-error.hh:25:
./mu-utils.hh:162:14: error: call to function 'operator<<' that is neither visible in the template definition nor found by argument-dependent lookup
        sstr << val;
             ^
mu-command-parser.cc:81:45: note: in instantiation of function template specialization 'Mu::to_string<Mu::Sexp::Type>' requested here
                                            to_string(arginfo.type) +
                                            ^
../../lib/utils/mu-sexp-parser.hh:85:1: note: 'operator<<' should be declared prior to the call site or in namespace 'Mu::Sexp'
operator<<(std::ostream& os, Sexp::Type id)
^
1 error generated.
make[4]: *** [mu-command-parser.lo] Error 1
make[3]: *** [all-recursive] Error 1
make[2]: *** [all] Error 2
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
```

**Comment by seanfarley** [comment]
Fixed in e119ac919b037d64753d457d687d175cb2ca6d2d! Thanks so much!

![image](https://user-images.githubusercontent.com/20887/73030740-1a8abf00-3def-11ea-93fe-2a884ef51358.png)
